gloss/sys_times.c

```diff
@@ -3,7 +3,30 @@
 #include <sys/time.h>
 #include <sys/times.h>
 
-extern int _gettimeofday(struct timeval *, void *);
+/* v * num / den while avoiding overflow
+ *
+ * Modulus on unsigned values is defined as:
+ *
+ *	v % den = v - (v / den) * den
+ *	v = v % den + (v / den) * den
+ *
+ * This lets us break the computation down as follows:
+ *
+ *	r = (v * num) / den
+ *	  = ((v % den + (v / den) * den) * num) / den
+ *	  = ((v % den) * num) / den + ((v / den) * den) * num / den
+ *	  = ((v % den) * num) / den + ((v / den) * num
+ *
+ * As long as num * den fits in 64 bits, then this compuation will fit
```

Comment:
Good work, Keith!

To be pedantic, you might consider adding an obvious clarification,

> As long as num * den fits in 64 bits **and the (exact) result does too**

That is, this trick obviously cannot help if `v * num / den` overflows in exact arithmetic. (For example, take `v == 2`, `num == 2^64`, and `den == 1`.) 

There's also a typo: "compuation".

Reply:
Thanks for catching the typo; I'll get that fixed.

We don't actually need a * num / den to fit in 64 bits -- what we need is for the function to "work" for time computations. That just requires:
```
moddiv(a - b, num, den) = moddiv(a, num, den) - moddiv(b, num, den)
```
As this function computes the exact result of a * num / den, we'll always satisfy this, so
even if the result gets truncated to 64 bits, as long as the application operates on the results correctly, it will get the right result.

A reasonable clarification might just note that truncating the result to 64 bits will work fine.